packages/video_player/video_player_avfoundation/darwin/video_player_avfoundation/Sources/video_player_avfoundation/FVPVideoPlayerPlugin.m

```diff
@@ -234,9 +246,8 @@ - (AVMutableVideoComposition *)getVideoCompositionWithTransform:(CGAffineTransfo
   }
   videoComposition.renderSize = CGSizeMake(width, height);
 
-  // TODO(@recastrodiaz): should we use videoTrack.nominalFrameRate ?
-  // Currently set at a constant 30 FPS
-  videoComposition.frameDuration = CMTimeMake(1, 30);
+  videoComposition.sourceTrackIDForFrameTiming = videoTrack.trackID;
+  videoComposition.frameDuration = videoTrack.minFrameDuration;
```

Comment:
What happens if `minFrameDuration` is `kCMTimeInvalid`, which is a documented possibility?

Reply:
Setting `frameDuration` to `kCMTimeInvalid` causes an objc exception. But setting some magical constant as fallback also does not sound good, it can even be `CMTimeMake(1, 1)`. Maybe my thought at that time was that if it happens then someone will report it but with even bad fallback that is less probable. Because now we do not know why that could happen and what is incidence of that, if it happens only for some degenerate video files then magical constant may be ok but if it happens often then some more elaborate fix could be needed (like using display refresh rate, but which could be overkill if this does not happen at all). But maybe better would be to use some magical constant as fallback but also report it as error or warning.

From definition `minFrameDuration` should be defined if the video has at least two frames. I tested with video with just a single frame and `minFrameDuration` still had valid `CMTime` value, I tried to test with video without any video samples but that video sadly also did not contain any video tracks. I was only able to get `kCMTimeInvalid` in `minFrameDuration` when I used it on an audio track.

Reply:
Not everyone reports everything. Even if something is hard enough to notice does not mean users will not notice.

Reply:
I don't have any objection to once-per-video logging about this as long as it's:
- only compiled in debug mode, and
- has clear, actionable steps for a developer hitting it

Reply:
Currently it is `NSLog(@"Warning: videoTrack.minFrameDuration for input video is invalid, please report this.");`, that is why I asked what is typically used in packages for such logs.

Reply:
> that is why I asked what is typically used in packages for such logs

Typically we don't do logging like this, because it's very hard to notice; instead we return errors so they will be surfaced using the usual process.

In this case I don't think adding a new error return that didn't previously exist is worth the risk of regression though.